templates/examples/extensions/py_lambda_hooks/CanvasHook/CanvasHook.py

```diff
@@ -42,162 +47,150 @@ def get_secret(secrets_id_name):
-            # An error occurred on the server side.
-            # Deal with the exception here, and/or rethrow at your discretion.
-            raise e
-        elif e.response['Error']['Code'] == 'InvalidParameterException':
-            # You provided an invalid value for a parameter.
-            # Deal with the exception here, and/or rethrow at your discretion.
-            raise e
-        elif e.response['Error']['Code'] == 'InvalidRequestException':
-            # You provided a parameter value that is not valid for the current state of the resource.
-            # Deal with the exception here, and/or rethrow at your discretion.
-            raise e
-        elif e.response['Error']['Code'] == 'ResourceNotFoundException':
-            # We can't find the resource that you asked for.
-            # Deal with the exception here, and/or rethrow at your discretion.
-            raise e
-    else:
         # Decrypts secret using the associated KMS CMK.
         # Depending on whether the secret is a string or binary, one of these fields will be populated.
         if 'SecretString' in get_secret_value_response:
             secret = get_secret_value_response['SecretString']
+            secret = json.loads(get_secret_value_response['SecretString'])['API_Token']
         else:
             decoded_binary_secret = base64.b64decode(get_secret_value_response['SecretBinary'])
+            secret = decoded_binary_secret.API_Token
+    except ClientError as e:
+        print ("ERROR: "+ str(e))  #print the exception
+        raise str(e)
 
     #return the API token
-    return json.loads(get_secret_value_response['SecretString'])['API_Token']
-
+    return secret
 
-#----------------------------------------------------------------------
-# function to get Canvas User by using  user_id to match with LMS SIS_ID
-#----------------------------------------------------------------------
 
+"""
+function to get Canvas User
+This function retrieves the Canvas user by using the SIS Login ID
+"""
 def getCanvasUser (param_canvas, param_user_name):
     user = param_canvas.get_user(param_user_name, 'sis_login_id')
     return user
 
-#----------------------------------------------------------------------
-# function: query_enrollments_for_student
-#----------------------------------------------------------------------
-def query_enrollments_for_student(canvas, student_user_name, userinput):
-    enrollments_for_student = 'You are not currently enrolled in any courses.'
 
+"""
+function: query_enrollments_for_student
+This function retrieves students' active enrollments 
+"""
+def query_enrollments_for_student(canvas, student_user_name, userinput):
     # Get the user using user_id to match with LMS SIS_ID
     user = getCanvasUser (canvas, student_user_name)
 
     if user:
         courses = user.get_courses(enrollment_status='active',include=['syllabus_body'])
 
-        course_name = []
+        course_names = []
```

Comment:
For list comprehension, the `course_name ... if course` is not really needed, this can e changed to a one-liner

```course_names = [course.name for course in courses]```